### PR TITLE
Scale XBLOCK in triton for `pointwise`

### PR DESCRIPTION
**Pull request by oraluben**
Adjust https://github.com/pytorch/pytorch/pull/128826 for also `triton_heuristics.pointwise`.

An example we encountered during training qwen-7b with rocm 6.1:

Note: this kernel also hit the limit of `TRITON_MAX_BLOCK['X']`, shall we increase it from 2048 to 4096?

```

import torch

aten = torch.ops.aten
inductor_ops = torch.ops.inductor
assert_size_stride = torch._C._dynamo.guards.assert_size_stride
empty_strided_cpu = torch._C._dynamo.guards._empty_strided_cpu
empty_strided_cuda = torch._C._dynamo.guards._empty_strided_cuda
reinterpret_tensor = torch._C._dynamo.guards._reinterpret_tensor
alloc_from_pool = torch.ops.inductor._alloc_from_pool


import triton
import triton.language as tl
from triton.compiler.compiler import AttrsDescriptor

from torch._inductor.runtime import triton_heuristics
from torch._inductor.runtime.hints import DeviceProperties

@triton_heuristics.pointwise(
    size_hints=[8589934592], 
    filename=__file__,
    triton_meta={'signature': {0: '*bf16'}, 'device': DeviceProperties(type='hip', index=2, cc='gfx942', major=None, regs_per_multiprocessor=None, max_threads_per_multi_processor=None, multi_processor_count=None), 'constants': {}, 'configs': [AttrsDescriptor(divisible_by_16=(0, 1), equal_to_1=())]},
    inductor_meta={'autotune_hints': set(), 'kernel_name': 'triton_poi_fused_nll_loss_backward_0', 'mutated_arg_names': [], 'no_x_dim': False, 'num_load': 0, 'num_reduction': 0, 'backend_hash': None, 'are_deterministic_algorithms_enabled': False, 'assert_indirect_indexing': True, 'autotune_local_cache': True, 'autotune_pointwise': True, 'autotune_remote_cache': False, 'force_disable_caches': False, 'dynamic_scale_rblock': True, 'max_autotune': False, 'max_autotune_pointwise': False, 'min_split_scan_rblock': 256, 'spill_threshold': 16, 'store_cubin': False, 'is_hip': True},
    min_elem_per_thread=0
)
@triton.jit
def triton_(out_ptr0, XBLOCK : tl.constexpr):
    xoffset = tl.program_id(0).to(tl.int64) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:].to(tl.int64)
    x0 = xindex
    tmp0 = 0.0
    tl.store(out_ptr0 + (x0), tmp0, None)

import triton
import triton.language as tl
from torch._inductor.runtime.triton_heuristics import grid
from torch._C import _cuda_getCurrentRawStream as get_raw_stream


if __name__ == "__main__":
    with torch.cuda._DeviceGuard(2):
        torch.cuda.set_device(2)
        buf0 = empty_strided_cuda((32752, 151936), (151936, 1), torch.bfloat16)
        stream2 = get_raw_stream(2)
        triton_.run(buf0, grid=grid(4976207872), stream=stream2)

```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jansel — CHANGES_REQUESTED** [review]
Please fix lints and tests

**Review by jansel — APPROVED** [review]

**Comment by oraluben** [comment]
@jansel would you mind helping to merge this? it seems to blocked by a unrelated failure and the test result is not updated.

**Comment by jansel** [comment]
@oraluben looks like this branch has merge conflicts, which you need to handle locally.

Please rebase this against the `viable/strict` branch and update the PR.

**Comment by oraluben** [comment]
@jansel it needs another approval to run the ci : )

File: torch/_inductor/runtime/triton_heuristics.py
Comment (Chillee) [inline torch/_inductor/runtime/triton_heuristics.py:1280]: Why is this scaling it based off of `num_blocks * num_warps * warp_size` instead of just `num_blocks`?

Comment (Chillee) [reply torch/_inductor/runtime/triton_heuristics.py:1280]: This is causing issues due to https://github.com/pytorch/pytorch/issues/135028

File: torch/_inductor/runtime/triton_heuristics.py
Comment (Chillee) [inline torch/_inductor/runtime/triton_heuristics.py:1283]: `x` is the block size. Why are we erroring if it's greater then the maximum grid size?

Comment (oraluben) [reply torch/_inductor/runtime/triton_heuristics.py:1283]: Does this make more sense to you?
```
diff --git a/torch/_inductor/runtime/triton_heuristics.py b/torch/_inductor/runtime/triton_heuristics.py
index c5a6bd69fd7..db034e2d14a 100644
--- a/torch/_inductor/runtime/triton_heuristics.py
+++ b/torch/_inductor/runtime/triton_heuristics.py
@@ -1139,10 +1139,10 @@ def _check_max_grid_x(size_hints, x, num_warps):
     while (num_blocks * num_warps * warp_size) > max_grid_x and x < size_hints[0]:
         x *= 2  # Scale up XBLOCK if grid exceeds limits
         num_blocks = num_blocks // 2
-        if x >= max_grid_x:
-            raise AssertionError(
-                "Reduction config exceeds cudaDeviceProp maxGridSize. Please raise a pytorch issue"
-            )
+    if (num_blocks * num_warps * warp_size) > max_grid_x:
+        raise AssertionError(
+            "Reduction config exceeds cudaDeviceProp maxGridSize. Please raise a pytorch issue"
+        )
     return x, num_blocks
```

Comment (Chillee) [reply torch/_inductor/runtime/triton_heuristics.py:1283]: Yeah, that makes more sense. But i still don't understand the overall constraint here we're trying to solve. 

The other thing here is that we're scaling up here without respecting the original constraints in the pointwise op (i.e.  `size_hints/block_x < maxGridSize`).

It would be preferable to put these constraints all in one place.

**Pull request closed without merging**
